wasm_test.js

```diff
@@ -0,0 +1,51 @@
+// Simple smoke test for WASM mypy wheels
+const { loadPyodide } = require("pyodide");
+const process = require("process");
+const fs = require("fs");
+const path = require("path");
+
+
+const EXPECTED_ERROR = `test.py:1: error: Unsupported operand types for + ("int" and "str")  [operator]
+Found 1 error in 1 file (checked 1 source file)
+`
+
+
+async function test_mypy() {
+  let pyodide = await loadPyodide();
+  await pyodide.loadPackage("micropip")
+  let micropip = pyodide.pyimport("micropip");
+  let requirements = fs.readFileSync("mypy/mypy-requirements.txt", {encoding: "utf8"});
+  await Promise.all(
+    requirements.split("\n").map(async (requirement) => {
+      if (!requirement.startsWith("#") && requirement != "") {
+        await micropip.install([requirement]);
+      }
+    })
+  );
+  const dist_dir = "mypy/dist";
+  const files = await fs.promises.readdir(dist_dir);
+  await pyodide.loadPackage(path.join(dist_dir, files[0]));
+  pyodide.runPython(`
+    with open('test.py', 'w') as f:
+        f.write("1 + ''")
+  `);
+  return pyodide.runPython("import mypy.api; mypy.api.run(['test.py'])");
+}
```

Comment:
Not quite sure about how pyodide's import system works...
- Will `import x` not import `x.py` unless loadPackage was previously called? If not, maybe we want to move to a directory that doesn't have mypy in it just to make sure we're definitely getting the wheel?
- Would we get better coverage in any way if we used `micropip` to install the `.whl` file instead of loadPackage (if even possible)?

Reply:
The pyodide/Python runtime is running in an Emscripten virtual filesystem, so it is sandboxed from the host filesystem. So I don't believe there is risk we are accidentally picking up the `mypy` folder (I don't think we'd pick it up anyway, the script is run from the top-level and so it'd have to recurse through mypy's checkout).

As for `micropip` vs `loadPackage`, my understanding is that if `micropip` sees `.whl`, it hands things off to `loadPackage`, so I don't think there is better coverage.